x/feedistribution/keeper/msg_update_params.go

```diff
@@ -0,0 +1,21 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/ExocoreNetwork/exocore/x/feedistribution/types"
+)
+
+func (k msgServer) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams) (*types.MsgUpdateParamsResponse, error) {
+	// For test purpose, skip for now
+	// if k.GetAuthority() != req.Authority {
+	// 	 return nil, errorsmod.Wrapf(types.ErrInvalidSigner, "invalid authority; expected %s, got %s", k.GetAuthority(), req.Authority)
+	// }
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	k.SetParams(ctx, req.Params)
```

Comment:
Check that the `EpochIdentifier` is registered in the `x/epochs` module

Reply:
Not sure I understand your response. The `MsgUpdateParams` could contain an invalid epoch not registered within the `x/epochs` module. For example, change of the epoch identifier to `max` should be deemed invalid.